File: backend/app/routers/analytics.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database.connection import get_db
from app.services.analytics_service import AnalyticsService

router = APIRouter(
    prefix="/analytics",
    tags=["Analytics"]
)


@router.get("/summary")
def summary(db: Session = Depends(get_db)):
    return AnalyticsService.dashboard_summary(db)


@router.get("/congestion")
def congestion(db: Session = Depends(get_db)):
    return AnalyticsService.congestion_distribution(db)


@router.get("/busiest")
def busiest(db: Session = Depends(get_db)):
    return AnalyticsService.busiest_roads(db)


@router.get("/fastest")
def fastest(db: Session = Depends(get_db)):
    return AnalyticsService.fastest_roads(db)


@router.get("/trend")
def traffic_trend(db: Session = Depends(get_db)):
    return AnalyticsService.traffic_trend(db)


@router.get("/insights")
def insights(db: Session = Depends(get_db)):
    return AnalyticsService.ai_insights(db)

File: frontend/src/services/analytics.js
import api from "./api";

export async function getDashboardSummary() {
  const response = await api.get("/analytics/summary");
  return response.data;
}

export async function getCongestionChart() {
  const response = await api.get("/analytics/congestion");
  return response.data;
}

export async function getBusiestRoads() {
  const response = await api.get("/analytics/busiest");
  return response.data;
}

export async function getFastestRoads() {
  const response = await api.get("/analytics/fastest");
  return response.data;
}

export const getTrafficTrend = async () => {
    const response = await api.get("/analytics/trend");
    return response.data;
};

export const getAIInsights = async () => {
    const response = await api.get("/analytics/insights");
    return response.data;
};